docs(webhook): Inventory webhook description. (#2470)
mendersoftware/mender-docs: 08.Server-integration/07.Webhooks/docs.md

    category: docs
---

Mender supports Webhooks to send data to third-party systems. With this mechanism, it's possible

> to notify external applications about device lifecycle and inventory changed events. This helps
> you avoid polling and manual synchronization between Mender and other systems, as well as allows
> you to handle device-related events and integrate Mender into your ecosystem seamlessly.

Reviewer: ```suggestion
to notify external applications about device lifecycle and inventory changed events. This eliminates the need for manual polling and synchronization, allowing for efficient handling of device-related events.
```

Author: fixed. I only could not resist and added extra sentence about integration.

Reviewer: Most of my English teachers were not pleased with my brevity. But I hold on to the one who gave me feedback that said she liked how I would ruthlessly cut out things. I guess it's a bit of cognitive dissonance avoidance or confirmation bias. :P

We divided webhook integrations into two categories (_scopes_): `deviceauth` and `inventory`.

> The former is available in all Mender plans, as well as Mender Open Source,

Reviewer: I want to suggest a link to mender plans here when mentioned. I dunno how stable that url is. https://mender.io/pricing/plans

Author: sure.

### External service

> You need an external service or application exposed on the internet and able to receive data
> from Mender in the form of an HTTP POST request. The hostname part of the URL must resolve
> to a globally routable unicast IP address. We recommend using HTTPS and configuring
> a secret for request signing purposes (see [below](#signature-header)).

Reviewer: ```suggestion
To receive data from Mender, you'll need an internet-accessible service or app that can accept HTTP POST requests. The service's domain name should resolve to a publicly accessible IP address. For security, we recommend using HTTPS and setting up a secret for verifying request authenticity (see [below](#signature-header)).
```

Author: fixed. I just added `unicast`. I like that word. Unicast.

## Supported events

### Device lifecycle (deviceauth) triggers

A webhook in the `deviceauth` scope fires on device lifecycle events:
* device provisioning,
* device decommissioning, or
* its authentication status changes.

### Inventory changed triggers

> A webhook in the `inventory` scope fires when device changed inventory events:

Reviewer: ```suggestion
A webhook in the `inventory` scope fires whenever Mender Server detects and updates a change in a device's inventory information.
```

> * Mender Server detected a change in the device inventory and updated it in the db

Reviewer: ```suggestion
```

Author: removed.

          "RelaxedOS",
          "Based GNU/Linux 11 (bullseye)"
        ],
        "scope": "inventory"
      },
      {
        "name": "rootfs-image.checksum",
        "value": "dc51b8c96c2d745df3bd5590d990230a482fd247123599548e0632fdbf97fc22",
        "scope": "inventory"
      },
      {
        "name": "rootfs-image.large.version",
        "value": "1.0.0",
        "scope": "inventory"
      },
      {
        "name": "rootfs-image.version",
        "value": "2023-05-03-raspios-bullseye-armhf-lite-mender",
        "scope": "inventory"
      },
      {
        "name": "rootfs_type",
        "value": "ext4",
        "scope": "inventory"
      },
      {
        "name": "update_modules",
        "value": [
          "deb",
          "directory",
          "docker",
          "mender-configure",
          "rootfs-image",
          "rpm",
          "script",
          "single-file"
        ],
        "scope": "inventory"
      }
    ]
  },
  "time": "2024-10-04T08:13:03.570584412Z"
}
```

More information about payload possibilities are available
in the [iot-manager](https://github.com/mendersoftware/iot-manager) management API specification.

## Signature header

> If you specify a secret, an integrity check is calculated and located in the `X-Men-Signature` header,

Reviewer: Why `X-Men`?  Could be Avengers, Fantastic Four, Justice League, Teen Titans.

Author: haha! a. I like XMen second: can we make it Spiderman?
